src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -109,6 +110,13 @@ object LightGBMUtils {
     idAsInt
   }
 
+  def getHostToIP(hostname: String): String = {
+    if (InetAddresses.isInetAddress(hostname))
+      hostname
+    else
+      InetAddress.getByName(hostname).getAddress.map(_ & 0xFF).mkString(".")
```

Comment:
I think here you can directly use getHostAddress() function?

Reply:
done, I was thinking of using that too originally but this seemed more reliable since it returns the ints directly, but since it worked for your cluster I will use it